docs/how-to-guides/deploy.md

```diff
@@ -0,0 +1,279 @@
+  type = string
+  default = "YOUR_CREDENTIAL_SECRET"
+}
+
+variable "openstack_application_credential_secret" {
+  type = string
+  default = "YOUR_CREDENTIAL_SECRET"
+}
+```
+
+…filling in your credentials that you generated earlier, and making sure the
+`SSH_PUBLIC_KEY` path matches that of the accompanying public key to the
+`SSH_KEY` you supplied earlier.
+
+Lastly, in the `terraform/ansible/roles/softpack_builder/tasks/main.yml` file,
+**remove** the following lines:
+
+```
+- name: Import GPG key
+  vars:
+    key: /tmp/softpack.pgp
+  block:
+    - name: Copy GPG key
+      ansible.builtin.copy:
+        src: "{{ lookup('ansible.builtin.env', 'SPACK_KEY') }}"
+        dest: "{{ key }}"
+        mode: u=rw,g=-,o=-
+
+    - name: Import GPG key
+      ansible.builtin.shell: |
+        SPACK_GNUPGHOME={{ builder.gpg }} {{ spack_path }}/bin/spack gpg trust {{ key }}
+      args:
+        creates: "{{ builder.gpg }}/trustdb.gpg"
+```
+
+## Building the VM
+
+For this stage, you'll need to have both [Terraform](https://www.terraform.io/)
+and [Ansible](https://www.ansible.com/) installed. It is recommended to install
+`ansible` using `pip`:
+
+```
+pip install ansible
+```
+
+Once done, make sure that you're in the `infrastructure/terraform` directory,
+and run the following two commands to provision the VM for SoftPack:
+
+```
+terraform init
+terraform apply
+```
+
+…following the onscreen instructions to confirm the actions that will take
+place.
+
+## Installing the Software
+
+While still in the `infrastructure/terraform` directory, you can run the
+following command to set-up the newly provisioned VMs with the necessary
+software and configuration:
+
+```
+ansible-playbook --inventory ansible/hosts.yml ansible/main.yml
+```
+
+You will need to response 'yes' to each of the three SSH authenticity checks,
+but should otherwise not be required to interact with the process.
+
+NB: This will most likely take a few minutes.
+
+## After Installation Modification
+
+Once the initial software configuration is done, you'll need to log in to the
+machine to make some changes:
+
+```
+ssh ubuntu@builder.softpack-USERNAME.hgi-dev.sanger.ac.uk
+```
+
+Once logged in, you'll need to create a directory and modify some files.
+
+Create the needed directory with the following command:
+
+```
+mkdir /opt/softpack/gpg/
+```
+
+The first file to edit is in the softpack builder configuration file, located at
+`/opt/softpack/builder/softpack_builder/config/conf/config.yml`. In the `vault`
+section, You'll need to fill in the details for the URL, Path, and Token for the
+vault, either with real values, or dummy ones such as the following:
+
+```
+vault:
+  url: https://something.com
+  path: /something
+  token: something
+```
+
+For the file `~/.softpack/builder/config.yml`, in the
+`spack->manifest->spack->container->images` section, remove the following three
+lines:
+
+```
+        images:
+          os: ubuntu:20.04
```

Comment:
Why does this need to be removed? Would be good to have a brief explanation for this one.

Reply:
It needs to be removed as it conflicts with other config with explicity sets the build and final base images.

I don't actually know why this config is there at all; it may just be default config that needs removing.

Either way, I'll see about adding an explanation for this.